File: frontend/src/app/shared/services/documento.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { environment } from "../../../../environment";
import { Observable } from "rxjs";

@Injectable({
    providedIn: 'root'
})
export class DocumentoService {

    private baseUrl = `${environment.apiUrl}/documentos`

    constructor( private http: HttpClient) {

    }

    gerarDeclaracao(id: number) : Observable<Blob> {
        return this.http.get(`${this.baseUrl}/declaracao/${id}`, { responseType: 'blob'})
    }

    gerarMatricula(id: number) : Observable<Blob> {
        return this.http.get(`${this.baseUrl}/matricula/${id}`, { responseType: 'blob'})
    }

    gerarHistorico(id: number) : Observable<Blob> {
        return this.http.get(`${this.baseUrl}/historico/${id}`, { responseType: 'blob'})
    }
}

File: backend/src/main/java/br/com/stbp/service/DocumentoService.java
package br.com.stbp.service;

import org.springframework.core.io.Resource;
import org.springframework.http.ResponseEntity;


public interface DocumentoService {


    public ResponseEntity<Resource> gerarDeclaracao(Long id);

    public ResponseEntity<Resource> gerarHistorico(Long id);

    public ResponseEntity<Resource> gerarMatricula(Long id);
}
